pkgs/applications/graphics/krita/default.nix

```diff
@@ -44,6 +44,15 @@ mkDerivation rec {
     "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
   ];
 
+  postInstall = ''
+    mkdir -p $out/${python3Packages.python.sitePackages}
+    ln -s ${python3Packages.pyqt5}/${python3Packages.python.sitePackages}/PyQt5/ $out/${python3Packages.python.sitePackages}/PyQt5
+    export PYTHONPATH=$PYTHONPATH:$out/${python3Packages.python.sitePackages}
```

Comment:
Why all this?  I think the for-loop below would be sufficient.  I confirm that the error message does disappear (but I don't use Krita).

Reply:
I was under the impression that the pythonpath would not contain PyQt. 
After taking a look at the resulting startup script it turns out I'm wrong. 
The (sorted and formatted) pythonpath list looks like that:
```
/nix/store/6w864g2fjwx8x8vvbw8mp33z4r9n54qi-python3.7-sip-4.19.13/lib/python3.7/site-packages
/nix/store/6w864g2fjwx8x8vvbw8mp33z4r9n54qi-python3.7-sip-4.19.13/lib/python3.7/site-packages
/nix/store/6w864g2fjwx8x8vvbw8mp33z4r9n54qi-python3.7-sip-4.19.13/lib/python3.7/site-packages
/nix/store/6w864g2fjwx8x8vvbw8mp33z4r9n54qi-python3.7-sip-4.19.13/lib/python3.7/site-packages
/nix/store/6w864g2fjwx8x8vvbw8mp33z4r9n54qi-python3.7-sip-4.19.13/lib/python3.7/site-packages
/nix/store/6w864g2fjwx8x8vvbw8mp33z4r9n54qi-python3.7-sip-4.19.13/lib/python3.7/site-packages
/nix/store/6w864g2fjwx8x8vvbw8mp33z4r9n54qi-python3.7-sip-4.19.13/lib/python3.7/site-packages
/nix/store/6w864g2fjwx8x8vvbw8mp33z4r9n54qi-python3.7-sip-4.19.13/lib/python3.7/site-packages
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/j560xfxnbfnb6z5fks1gjyv987qv2kip-krita-4.1.8/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
```

There is a similar issue in the pyqt5 Package, it's pythonpath looks like that.
```
/nix/store/86w4n5knvwpfc4h1i1gn984hwrxxjj5n-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
/nix/store/imvzjph1rfizpq3acb7yp9lgjc68hk4d-python3.7-PyQt5.sip-4.19.13/lib/python3.7/site-packages
/nix/store/imvzjph1rfizpq3acb7yp9lgjc68hk4d-python3.7-PyQt5.sip-4.19.13/lib/python3.7/site-packages
/nix/store/imvzjph1rfizpq3acb7yp9lgjc68hk4d-python3.7-PyQt5.sip-4.19.13/lib/python3.7/site-packages
/nix/store/s3xn5ig3ygqsrpl9ycgm4jw4a89naj0j-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/s5f3vpmig33nk4zyk228q55wdydd3pc2-python3-3.7.3/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
```

We should probably also be concerned about the fact that the nativeBuildInputs and other build time dependencies end up in there.
I've removed the above mentioned lines and am currently rebuilding it right now (that will take a few  hours though).

Reply:
I tested by simply copying `bin/krita` to a temporary dir and hand-editing the wrapper not to contain this extra path.

Reply:
Having duplicates isn't nice etc, but that's not krita-specific at all...  if the closure blowup was too large, we _might_ consider not using this automatic list and only explicitly adding packages that are needed or something like that.

Reply:
Aren't nativeBuildInputs removed by a garbage collect?  If so, wouldn't that mean that some packages could in theory break because of that (or missbehave).

Seems fine to me (pythonpath is now one line shorter)... though for some reason krita segfaults on exit again... though for some reason it does so for the previous build as well, weird.

Reply:
Runtime dependencies are *discovered* by scanning for hashes when registering the resulting /nix/store paths.  There were some suggestions to disallow `nativeBuildInputs` from this, but I'm not sure how far this has gone, and anyway it's not a risk _here_ (unless we cared for cross-compiling krita, but that doesn't seem a likely use case).

Reply:
Some of the Qt stuff is a bit risky around mixing of active versions on one system IIRC, but I can only guess it could be related to the crashes.  I get some QML errors when I don't "install" Krita (via nix-env or systemPackages etc.), but I think that's how it is with our QML so far.